common/src/main/java/com/wynntils/models/characterstate/CharacterStateModel.java

```diff
@@ -105,4 +114,16 @@ public int getTicksToNextSoulPoint() {
     private void centerSegmentCleared() {
         powderSpecialSegment.replaced();
     }
+
+    @SubscribeEvent
+    public void onChatReceived(ChatMessageReceivedEvent e) {
+        if (!e.getCodedMessage().contains(WYNN_DEATH_MESSAGE)) return;
+        WynntilsMod.postEvent(new CharacterDeathEvent(lastLocation));
+    }
+
+    @SubscribeEvent
+    public void onTick(TickEvent e) {
+        if (McUtils.player() == null) return;
+        lastLocation = McUtils.player().position();
```

Comment:
Actually, come to think of it, you are probably better of using a `Location` than a `Position`. The difference is that the former are integer coordinates only (BlockPos), while the latter have float coordinates (like the actual position of an entity). Minecraft will give you a position, but you really only care about the closest block. 

Sorry for not realizing this before, but even better then is to replace all instances of `Position` (or `Vec3`) in this PR with `Location`, and store the lastLocation using `lastLocation = Location.containing(McUtils.player().position());`